scaaml/capture/aes_capture.py

```diff
@@ -0,0 +1,206 @@
+from pathlib import Path
+
+from scaaml.capture.aes_capture_util import *
+from scaaml.capture.acqktp import AcqKeyTextPattern_Scaaml as ktp_scaaml
+from scaaml.io import Dataset
+from scaaml.io.resume_kti import ResumeKTI
+
+
+class AESCaptureRunner:
```

Comment:
According to the parameters, this class is very specific to captures done using exclusively Chipwhisperer hardware (scope + UART + clock generator). This is not the case for SCAAML datasets where the clock of the chip is different from the clock of the scope.

This should instead consider that we pass 4 objects in constructor, each implementing a defined API (and in specific cases, these can be the same object instance indeed):
* a scope object (chipwhisperer, picoscope, etc.)
* a control object (turning on/off the chip, resetting the chip, etc.) -- can be `None` in case we don't have a way to control the target board
* a communication object (chipwhisperer UART, external UART, etc.) -- this is typically the `target` parameter in Chipwhisperer API
* a crypto algorithm, providing the attack points and maybe some basic information about it (e.g. key length, etc.)